### ISSUE TITLE
Bug (or feature) in the new Style/ExtraSpacing cop

### ISSUE THREAD

**Issue by bquorning** [bbatsov/rubocop#2084]
Using latest master (cdfb667243136b173ddbac804ded35a48d5ec53a) and the following file test.rb:
```rb
# encoding: utf-8

# Indentation is fine
puts(
  a:   'a',
  bbb: 'b',
  c:   'c'
)

# Indentation is off
puts(
  d:  'd',
  eee: 'e',
  f:   'f'
)

# Indentation is off
puts(
  g: 'g',
  hhh: 'h',
  i:   'i'
)
```
RuboCop tells me this:
```
test.rb:12:5: C: Unnecessary spacing detected.
  d:  'd',
    ^
```
So the inconsistent indentation of “d, eee, f” was detected, but the inconsistency of “g, hhh, i” was not.

I believe the cop sees “g” as not indented at all, and “hhh, i” as consistently indented, so probably this is actually a feature, and not a bug?

@jonas054

**Comment by jonas054** [comment]
Yes, I would say so. It's difficult to accurately detect what is alignment and what isn't, so I chose some simple rules in order to allow anything that is potentially extra spacing used for alignment. In the g/h/i case, `'h'` and `'i'` are aligned, which makes the extra space before `'i'` OK. The fact that `'g'`, which is not preceded by extra space, is not aligned to the same column is beyond the scope of the cop IMO.
